src/check/constrain/generate/control_flow.rs

```diff
@@ -39,7 +39,7 @@ pub fn gen_flow(
                 .raises_caught(&raises_before);
 
             constrain_cases(ast, cases, &outer_env, ctx, constr)?;
-            Ok(env.clone())
+            Ok(outer_env.clone())
```

Comment:
This logic should be implemented in it's own PR.
This was to make sure that variables defined in handle can be used after handle.